design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,182 @@
+---
+Service authentication with vault for RGW
+target-version: release-1.15
+---
+
+# Feature Name
+
+Service account authentication with vault for RGW.
+
+## Summary
+
+When `Vault` is set up in Kubernetes environment, it can be authenticated using service account token. The OSD encryption already supports this feature and adding same for RGW. It uses same set of  config option in `ConnectionDetails` for `KeyManagementServiceSpec`.
+
+### Goals
+
+Configure RGW with service account authentication with Vault with help of vault agent.
+
+### Non-Goals
+
+There are different ways RGW can authenticate with help of [vault agent](https://docs.ceph.com/en/latest/radosgw/vault/#vault-agent), but only service account authentication is supported here.
+
+## Proposal details
+
+The [vault agent](https://developer.hashicorp.com/vault/docs/agent-and-proxy/agent#vault-stanza) can be configure with help of following settings in `connectionDetails` in `CephObjectStore CRD` and it will set following annotations in rgw pods.
+
+* `VAULT_AGENT_ADDR`     --> provides address of vault agent
+* `VAULT_AGENT_CM`       --> if address is localhost then agent need to configured along with rgw pod passed with help of cm, role for k8s service account need to add here
+* `VAULT_AGENT_TLS_CERT` --> tls cert for vault agent to communicate with vault server
+
+If vault agent address points to localhost following annotation will set on rgw pod:
+
+* vault.hashicorp.com/agent-inject    --> brings vault agent with rgw pod
+* vault.hashicorp.com/agent-configmap -->  config map containing settings for vault agent pod
```

Comment:
I think it's okay if we allow users to set these annotations on the CephObjectStore themselves to inject the agent if they want the sidecar implementation.

But I do want to avoid Rook maintaining sidecar resource definitions.